api/python/t4/packages.py

```diff
@@ -189,20 +195,21 @@ def deserialize(self):
             hash verification fail
             when deserialization metadata is not present
         """
-        target_str = self.meta.get('target')
-        if target_str is None:
-            raise QuiltException("No serialization metadata")
-
-        try:
-            target = TargetType(target_str)
-        except ValueError:
-            raise QuiltException("Unknown serialization target: %r" % target_str)
-
         physical_key = _to_singleton(self.physical_keys)
         data, _ = get_bytes(physical_key)
+
+        if func is not None:
+            return func(data)
+
+        pkey_ext = pathlib.PurePosixPath(urlparse(unquote(physical_key)).path).suffix
```

Comment:
Fix this unquote, too: `unquote(urlparse(physical_key).path)`

Reply:
Ugh.  I've been awful about those.

Reply:
Fixed, checked full changeset for any others, found one, fixed that as well.